flight/Modules/OnScreenDisplay/osd_menu.c

```diff
@@ -1080,7 +1080,24 @@ const char * ir_protocol_names[HWBRAINRE1_IRPROTOCOL_MAXOPTVAL + 1] = {
 };
 
 #define MAX_ID_ILAP 9999999
-#define MAX_ID_TRACKMATE 9999
+#define MAX_ID_TRACKMATE 0xfff
+
+// check if this is a valid trackmate id
+// It seems like the ID has some weird requirements:
+// 1st nibble is 0, 2nd nibble is not 0, 1, 8, or F
+// 3rd and 4th nibbles are not 0 or F
+bool trackmate_valid(uint16_t trackmate_id)
+{
+	bool valid = ((trackmate_id & 0x0F00) >> 8) != 0x0;
+	valid = valid && (((trackmate_id & 0x0F00) >> 8) != 0x1);
+	valid = valid && (((trackmate_id & 0x0F00) >> 8) != 0x8);
+	valid = valid && (((trackmate_id & 0x0F00) >> 8) != 0xF);
+	valid = valid && (((trackmate_id & 0x00F0) >> 4) != 0x0);
+	valid = valid && (((trackmate_id & 0x00F0) >> 4) != 0xF);
+	valid = valid && ((trackmate_id & 0x000F) != 0x0);
+	valid = valid && ((trackmate_id & 0x000F) != 0xF);
```

Comment:
The compiler should do common subexpression elimination and short circuit evaluation, though the likely case is all of these are true and thus all need evaluation.